### DISCUSSION TITLE
How to deploy two monorepos apps to subdirectories

### DISCUSSION THREAD

**Discussion by JustinyAhin** [vercel/vercel#8662] [Help] [answered]
Hello,

I have a use case for which I can not find a solution with Vercel and I might need some help.

So I have two (SvelteKit) apps in a Turborepo with this structure: `/apps/app1` and `/apps/app2`.

I'd like to deploy them to the same domain name `example.com` like: `example.com/app1` and `example.com/app2`. But I can't find any proper documentation about how to do that. Anyone has an idea?

The domain name is from Namecheap.

Thanks :)

**Comment by amyegan** [comment, marked as answer]
Hi @JustinyAhin! We have a guide on [serving multiple projects under a single domain](https://vercel.com/guides/how-can-i-serve-multiple-projects-under-a-single-domain) that should help with this. You'll need to [add the custom domain](https://vercel.com/guides/how-do-i-add-a-custom-domain-to-my-vercel-project) to your account and use [rewrites](https://vercel.com/docs/project-configuration#project-configuration/rewrites) to make the separate apps looks like they are all deployed under the same root domain.

**Reply by ryu-liam** [reply]
Hi @amyegan! 
I am struggling with deploying two projects in the single domain. 
We added rewrites in vercel.json, the below code is the whole code in the file.
When I access to "https://project1.jp/" it works, but "https://project1.jp/lp" doesn't seem to redirect to project2.

```json
{
  "rewrites": [
    {
      "source": "/:match*",
      "destination": "https://project1.jp/:match*"
    },
    {
      "source": "/lp/:match*",
      "destination": "https://project2.vercel.app/lp/:match*"
    }
  ]
}
```

It's able to impletement it locally by using nginx on docker though.

```
server{
        listen 80;
        server_name localhost;

        location /lp {
            proxy_pass http://host.docker.internal:3001/lp;
        }
        location / {
            proxy_pass http://host.docker.internal:3000;
        }
  }
```

Thank you

**Reply by amyegan** [reply]
Hi @ryu-liam! I would recommend switching the order of your rewrites so the `/lp/:match*` one is checked first. The `/:match*` rewrite would actually match all paths, including any that start with `/lp`. I suspect that is picking up every path and never letting the rewrite fall through to the second project rewrite.

**Reply by ryu-liam** [reply]
@amyegan 
Thank you for the suggestions! 
I found out that the vercel.json wasn't in the root directory. The rewrite works after I replaced it correctly.

**Reply by chhuang** [reply]
Hi @amyegan, does this make the app slower to load because of the initial rewrite/redirection?

**Reply by GermaVinsmoke** [reply]
Does it work with turborepo also? @amyegan

**Reply by nooraldarraji** [reply]
`"So I have two (SvelteKit) apps in a Turborepo"` he mentioned that already
